// @aws-sdk/client-cloudformation/dist-es/commands/BatchDescribeTypeConfigurationsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchDescribeTypeConfigurationsInput, BatchDescribeTypeConfigurationsOutput } from "../models/models_0";
import { deserializeAws_queryBatchDescribeTypeConfigurationsCommand, serializeAws_queryBatchDescribeTypeConfigurationsCommand, } from "../protocols/Aws_query";
var BatchDescribeTypeConfigurationsCommand = (function (_super) {
    __extends(BatchDescribeTypeConfigurationsCommand, _super);
    function BatchDescribeTypeConfigurationsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BatchDescribeTypeConfigurationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "BatchDescribeTypeConfigurationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchDescribeTypeConfigurationsInput.filterSensitiveLog,
            outputFilterSensitiveLog: BatchDescribeTypeConfigurationsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchDescribeTypeConfigurationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryBatchDescribeTypeConfigurationsCommand(input, context);
    };
    BatchDescribeTypeConfigurationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryBatchDescribeTypeConfigurationsCommand(output, context);
    };
    return BatchDescribeTypeConfigurationsCommand;
}($Command));
export { BatchDescribeTypeConfigurationsCommand };
