// @aws-sdk/client-cloudformation/dist-es/commands/DeleteStackInstancesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteStackInstancesInput, DeleteStackInstancesOutput } from "../models/models_0";
import { deserializeAws_queryDeleteStackInstancesCommand, serializeAws_queryDeleteStackInstancesCommand, } from "../protocols/Aws_query";
var DeleteStackInstancesCommand = (function (_super) {
    __extends(DeleteStackInstancesCommand, _super);
    function DeleteStackInstancesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteStackInstancesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "DeleteStackInstancesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteStackInstancesInput.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteStackInstancesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteStackInstancesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteStackInstancesCommand(input, context);
    };
    DeleteStackInstancesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteStackInstancesCommand(output, context);
    };
    return DeleteStackInstancesCommand;
}($Command));
export { DeleteStackInstancesCommand };
