// @aws-sdk/client-cloudformation/dist-es/commands/DeleteStackCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteStackInput } from "../models/models_0";
import { deserializeAws_queryDeleteStackCommand, serializeAws_queryDeleteStackCommand } from "../protocols/Aws_query";
var DeleteStackCommand = (function (_super) {
    __extends(DeleteStackCommand, _super);
    function DeleteStackCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteStackCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "DeleteStackCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteStackInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteStackCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteStackCommand(input, context);
    };
    DeleteStackCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteStackCommand(output, context);
    };
    return DeleteStackCommand;
}($Command));
export { DeleteStackCommand };
