// @aws-sdk/client-cloudformation/dist-es/commands/DeregisterTypeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeregisterTypeInput, DeregisterTypeOutput } from "../models/models_0";
import { deserializeAws_queryDeregisterTypeCommand, serializeAws_queryDeregisterTypeCommand, } from "../protocols/Aws_query";
var DeregisterTypeCommand = (function (_super) {
    __extends(DeregisterTypeCommand, _super);
    function DeregisterTypeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeregisterTypeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "DeregisterTypeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeregisterTypeInput.filterSensitiveLog,
            outputFilterSensitiveLog: DeregisterTypeOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeregisterTypeCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeregisterTypeCommand(input, context);
    };
    DeregisterTypeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeregisterTypeCommand(output, context);
    };
    return DeregisterTypeCommand;
}($Command));
export { DeregisterTypeCommand };
