// @aws-sdk/client-cloudformation/dist-es/commands/DescribePublisherCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribePublisherInput, DescribePublisherOutput } from "../models/models_0";
import { deserializeAws_queryDescribePublisherCommand, serializeAws_queryDescribePublisherCommand, } from "../protocols/Aws_query";
var DescribePublisherCommand = (function (_super) {
    __extends(DescribePublisherCommand, _super);
    function DescribePublisherCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribePublisherCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "DescribePublisherCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribePublisherInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribePublisherOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribePublisherCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribePublisherCommand(input, context);
    };
    DescribePublisherCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribePublisherCommand(output, context);
    };
    return DescribePublisherCommand;
}($Command));
export { DescribePublisherCommand };
