// @aws-sdk/client-cloudformation/dist-es/commands/DescribeChangeSetHooksCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeChangeSetHooksInput, DescribeChangeSetHooksOutput } from "../models/models_0";
import { deserializeAws_queryDescribeChangeSetHooksCommand, serializeAws_queryDescribeChangeSetHooksCommand, } from "../protocols/Aws_query";
var DescribeChangeSetHooksCommand = (function (_super) {
    __extends(DescribeChangeSetHooksCommand, _super);
    function DescribeChangeSetHooksCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeChangeSetHooksCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "DescribeChangeSetHooksCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeChangeSetHooksInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeChangeSetHooksOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeChangeSetHooksCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeChangeSetHooksCommand(input, context);
    };
    DescribeChangeSetHooksCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeChangeSetHooksCommand(output, context);
    };
    return DescribeChangeSetHooksCommand;
}($Command));
export { DescribeChangeSetHooksCommand };
