// @aws-sdk/client-cloudformation/dist-es/commands/DescribeStackEventsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeStackEventsInput, DescribeStackEventsOutput } from "../models/models_0";
import { deserializeAws_queryDescribeStackEventsCommand, serializeAws_queryDescribeStackEventsCommand, } from "../protocols/Aws_query";
var DescribeStackEventsCommand = (function (_super) {
    __extends(DescribeStackEventsCommand, _super);
    function DescribeStackEventsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeStackEventsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "DescribeStackEventsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeStackEventsInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeStackEventsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeStackEventsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeStackEventsCommand(input, context);
    };
    DescribeStackEventsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeStackEventsCommand(output, context);
    };
    return DescribeStackEventsCommand;
}($Command));
export { DescribeStackEventsCommand };
