// @aws-sdk/client-cloudformation/dist-es/commands/DescribeAccountLimitsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeAccountLimitsInput, DescribeAccountLimitsOutput } from "../models/models_0";
import { deserializeAws_queryDescribeAccountLimitsCommand, serializeAws_queryDescribeAccountLimitsCommand, } from "../protocols/Aws_query";
var DescribeAccountLimitsCommand = (function (_super) {
    __extends(DescribeAccountLimitsCommand, _super);
    function DescribeAccountLimitsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeAccountLimitsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "DescribeAccountLimitsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeAccountLimitsInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeAccountLimitsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeAccountLimitsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeAccountLimitsCommand(input, context);
    };
    DescribeAccountLimitsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeAccountLimitsCommand(output, context);
    };
    return DescribeAccountLimitsCommand;
}($Command));
export { DescribeAccountLimitsCommand };
