// @aws-sdk/client-cloudformation/dist-es/commands/DescribeStacksCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeStacksInput, DescribeStacksOutput } from "../models/models_0";
import { deserializeAws_queryDescribeStacksCommand, serializeAws_queryDescribeStacksCommand, } from "../protocols/Aws_query";
var DescribeStacksCommand = (function (_super) {
    __extends(DescribeStacksCommand, _super);
    function DescribeStacksCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeStacksCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "DescribeStacksCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeStacksInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeStacksOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeStacksCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeStacksCommand(input, context);
    };
    DescribeStacksCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeStacksCommand(output, context);
    };
    return DescribeStacksCommand;
}($Command));
export { DescribeStacksCommand };
