// @aws-sdk/client-cloudformation/dist-es/commands/DescribeStackSetCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeStackSetInput, DescribeStackSetOutput } from "../models/models_0";
import { deserializeAws_queryDescribeStackSetCommand, serializeAws_queryDescribeStackSetCommand, } from "../protocols/Aws_query";
var DescribeStackSetCommand = (function (_super) {
    __extends(DescribeStackSetCommand, _super);
    function DescribeStackSetCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeStackSetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "DescribeStackSetCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeStackSetInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeStackSetOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeStackSetCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeStackSetCommand(input, context);
    };
    DescribeStackSetCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeStackSetCommand(output, context);
    };
    return DescribeStackSetCommand;
}($Command));
export { DescribeStackSetCommand };
