// @aws-sdk/client-cloudformation/dist-es/commands/DescribeTypeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeTypeInput, DescribeTypeOutput } from "../models/models_0";
import { deserializeAws_queryDescribeTypeCommand, serializeAws_queryDescribeTypeCommand } from "../protocols/Aws_query";
var DescribeTypeCommand = (function (_super) {
    __extends(DescribeTypeCommand, _super);
    function DescribeTypeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeTypeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "DescribeTypeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeTypeInput.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeTypeOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeTypeCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeTypeCommand(input, context);
    };
    DescribeTypeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeTypeCommand(output, context);
    };
    return DescribeTypeCommand;
}($Command));
export { DescribeTypeCommand };
