// @aws-sdk/client-cloudformation/dist-es/commands/DetectStackDriftCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DetectStackDriftInput, DetectStackDriftOutput } from "../models/models_0";
import { deserializeAws_queryDetectStackDriftCommand, serializeAws_queryDetectStackDriftCommand, } from "../protocols/Aws_query";
var DetectStackDriftCommand = (function (_super) {
    __extends(DetectStackDriftCommand, _super);
    function DetectStackDriftCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DetectStackDriftCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "DetectStackDriftCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DetectStackDriftInput.filterSensitiveLog,
            outputFilterSensitiveLog: DetectStackDriftOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DetectStackDriftCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDetectStackDriftCommand(input, context);
    };
    DetectStackDriftCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDetectStackDriftCommand(output, context);
    };
    return DetectStackDriftCommand;
}($Command));
export { DetectStackDriftCommand };
