// @aws-sdk/client-cloudformation/dist-es/commands/ExecuteChangeSetCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ExecuteChangeSetInput, ExecuteChangeSetOutput } from "../models/models_0";
import { deserializeAws_queryExecuteChangeSetCommand, serializeAws_queryExecuteChangeSetCommand, } from "../protocols/Aws_query";
var ExecuteChangeSetCommand = (function (_super) {
    __extends(ExecuteChangeSetCommand, _super);
    function ExecuteChangeSetCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ExecuteChangeSetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "ExecuteChangeSetCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ExecuteChangeSetInput.filterSensitiveLog,
            outputFilterSensitiveLog: ExecuteChangeSetOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ExecuteChangeSetCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryExecuteChangeSetCommand(input, context);
    };
    ExecuteChangeSetCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryExecuteChangeSetCommand(output, context);
    };
    return ExecuteChangeSetCommand;
}($Command));
export { ExecuteChangeSetCommand };
