// @aws-sdk/client-cloudformation/dist-es/commands/ListImportsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListImportsInput, ListImportsOutput } from "../models/models_0";
import { deserializeAws_queryListImportsCommand, serializeAws_queryListImportsCommand } from "../protocols/Aws_query";
var ListImportsCommand = (function (_super) {
    __extends(ListImportsCommand, _super);
    function ListImportsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListImportsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "ListImportsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListImportsInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListImportsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListImportsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListImportsCommand(input, context);
    };
    ListImportsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListImportsCommand(output, context);
    };
    return ListImportsCommand;
}($Command));
export { ListImportsCommand };
