// @aws-sdk/client-cloudformation/dist-es/commands/ListStackInstancesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListStackInstancesInput, ListStackInstancesOutput } from "../models/models_0";
import { deserializeAws_queryListStackInstancesCommand, serializeAws_queryListStackInstancesCommand, } from "../protocols/Aws_query";
var ListStackInstancesCommand = (function (_super) {
    __extends(ListStackInstancesCommand, _super);
    function ListStackInstancesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListStackInstancesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "ListStackInstancesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListStackInstancesInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListStackInstancesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListStackInstancesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListStackInstancesCommand(input, context);
    };
    ListStackInstancesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListStackInstancesCommand(output, context);
    };
    return ListStackInstancesCommand;
}($Command));
export { ListStackInstancesCommand };
