// @aws-sdk/client-cloudformation/dist-es/commands/ListStacksCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListStacksInput, ListStacksOutput } from "../models/models_0";
import { deserializeAws_queryListStacksCommand, serializeAws_queryListStacksCommand } from "../protocols/Aws_query";
var ListStacksCommand = (function (_super) {
    __extends(ListStacksCommand, _super);
    function ListStacksCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListStacksCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "ListStacksCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListStacksInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListStacksOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListStacksCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListStacksCommand(input, context);
    };
    ListStacksCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListStacksCommand(output, context);
    };
    return ListStacksCommand;
}($Command));
export { ListStacksCommand };
