// @aws-sdk/client-cloudformation/dist-es/commands/ListTypeRegistrationsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTypeRegistrationsInput, ListTypeRegistrationsOutput } from "../models/models_0";
import { deserializeAws_queryListTypeRegistrationsCommand, serializeAws_queryListTypeRegistrationsCommand, } from "../protocols/Aws_query";
var ListTypeRegistrationsCommand = (function (_super) {
    __extends(ListTypeRegistrationsCommand, _super);
    function ListTypeRegistrationsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListTypeRegistrationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "ListTypeRegistrationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTypeRegistrationsInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListTypeRegistrationsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTypeRegistrationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListTypeRegistrationsCommand(input, context);
    };
    ListTypeRegistrationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListTypeRegistrationsCommand(output, context);
    };
    return ListTypeRegistrationsCommand;
}($Command));
export { ListTypeRegistrationsCommand };
