// @aws-sdk/client-cloudformation/dist-es/commands/ListTypesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTypesInput, ListTypesOutput } from "../models/models_0";
import { deserializeAws_queryListTypesCommand, serializeAws_queryListTypesCommand } from "../protocols/Aws_query";
var ListTypesCommand = (function (_super) {
    __extends(ListTypesCommand, _super);
    function ListTypesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListTypesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "ListTypesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTypesInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListTypesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTypesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListTypesCommand(input, context);
    };
    ListTypesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListTypesCommand(output, context);
    };
    return ListTypesCommand;
}($Command));
export { ListTypesCommand };
