// @aws-sdk/client-cloudformation/dist-es/commands/ListChangeSetsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListChangeSetsInput, ListChangeSetsOutput } from "../models/models_0";
import { deserializeAws_queryListChangeSetsCommand, serializeAws_queryListChangeSetsCommand, } from "../protocols/Aws_query";
var ListChangeSetsCommand = (function (_super) {
    __extends(ListChangeSetsCommand, _super);
    function ListChangeSetsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListChangeSetsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "ListChangeSetsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListChangeSetsInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListChangeSetsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListChangeSetsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListChangeSetsCommand(input, context);
    };
    ListChangeSetsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListChangeSetsCommand(output, context);
    };
    return ListChangeSetsCommand;
}($Command));
export { ListChangeSetsCommand };
