// @aws-sdk/client-cloudformation/dist-es/commands/PublishTypeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PublishTypeInput, PublishTypeOutput } from "../models/models_0";
import { deserializeAws_queryPublishTypeCommand, serializeAws_queryPublishTypeCommand } from "../protocols/Aws_query";
var PublishTypeCommand = (function (_super) {
    __extends(PublishTypeCommand, _super);
    function PublishTypeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PublishTypeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "PublishTypeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PublishTypeInput.filterSensitiveLog,
            outputFilterSensitiveLog: PublishTypeOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PublishTypeCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryPublishTypeCommand(input, context);
    };
    PublishTypeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryPublishTypeCommand(output, context);
    };
    return PublishTypeCommand;
}($Command));
export { PublishTypeCommand };
