// @aws-sdk/client-cloudformation/dist-es/commands/RegisterPublisherCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RegisterPublisherInput, RegisterPublisherOutput } from "../models/models_0";
import { deserializeAws_queryRegisterPublisherCommand, serializeAws_queryRegisterPublisherCommand, } from "../protocols/Aws_query";
var RegisterPublisherCommand = (function (_super) {
    __extends(RegisterPublisherCommand, _super);
    function RegisterPublisherCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RegisterPublisherCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "RegisterPublisherCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RegisterPublisherInput.filterSensitiveLog,
            outputFilterSensitiveLog: RegisterPublisherOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RegisterPublisherCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRegisterPublisherCommand(input, context);
    };
    RegisterPublisherCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRegisterPublisherCommand(output, context);
    };
    return RegisterPublisherCommand;
}($Command));
export { RegisterPublisherCommand };
