// @aws-sdk/client-cloudformation/dist-es/commands/SignalResourceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SignalResourceInput } from "../models/models_0";
import { deserializeAws_querySignalResourceCommand, serializeAws_querySignalResourceCommand, } from "../protocols/Aws_query";
var SignalResourceCommand = (function (_super) {
    __extends(SignalResourceCommand, _super);
    function SignalResourceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SignalResourceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "SignalResourceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SignalResourceInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SignalResourceCommand.prototype.serialize = function (input, context) {
        return serializeAws_querySignalResourceCommand(input, context);
    };
    SignalResourceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_querySignalResourceCommand(output, context);
    };
    return SignalResourceCommand;
}($Command));
export { SignalResourceCommand };
