// @aws-sdk/client-cloudformation/dist-es/commands/UpdateTerminationProtectionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateTerminationProtectionInput, UpdateTerminationProtectionOutput } from "../models/models_0";
import { deserializeAws_queryUpdateTerminationProtectionCommand, serializeAws_queryUpdateTerminationProtectionCommand, } from "../protocols/Aws_query";
var UpdateTerminationProtectionCommand = (function (_super) {
    __extends(UpdateTerminationProtectionCommand, _super);
    function UpdateTerminationProtectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateTerminationProtectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "UpdateTerminationProtectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateTerminationProtectionInput.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateTerminationProtectionOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateTerminationProtectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUpdateTerminationProtectionCommand(input, context);
    };
    UpdateTerminationProtectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUpdateTerminationProtectionCommand(output, context);
    };
    return UpdateTerminationProtectionCommand;
}($Command));
export { UpdateTerminationProtectionCommand };
