// @aws-sdk/client-cloudformation/dist-es/commands/ValidateTemplateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ValidateTemplateInput, ValidateTemplateOutput } from "../models/models_0";
import { deserializeAws_queryValidateTemplateCommand, serializeAws_queryValidateTemplateCommand, } from "../protocols/Aws_query";
var ValidateTemplateCommand = (function (_super) {
    __extends(ValidateTemplateCommand, _super);
    function ValidateTemplateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ValidateTemplateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudFormationClient";
        var commandName = "ValidateTemplateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ValidateTemplateInput.filterSensitiveLog,
            outputFilterSensitiveLog: ValidateTemplateOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ValidateTemplateCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryValidateTemplateCommand(input, context);
    };
    ValidateTemplateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryValidateTemplateCommand(output, context);
    };
    return ValidateTemplateCommand;
}($Command));
export { ValidateTemplateCommand };
